pkg/volume/util/fs/fs.go

```diff
@@ -66,9 +72,18 @@ func DiskUsage(path string) (*resource.Quantity, error) {
 	} else if err != nil {
 		return nil, fmt.Errorf("unable to retrieve disk consumption via quota for %s: %v", path, err)
 	}
+
+	// Create a new context and add a timeout to it.
+	ctx, cancel := context.WithTimeout(context.Background(), duTimeout*time.Second)
+	defer cancel()
+
 	// Uses the same niceness level as cadvisor.fs does when running du
 	// Uses -B 1 to always scale to a blocksize of 1 byte
-	out, err := exec.Command("nice", "-n", "19", "du", "-x", "-s", "-B", "1", path).CombinedOutput()
+	out, err := exec.CommandContext(ctx, "nice", "-n", "19", "du", "-x", "-s", "-B", "1", path).CombinedOutput()
```

Comment:
I think a better fix would be to limit how many du processes kubelet can spawn at a time.

Reply:
i agree with @msau42

Reply:
> I think a better fix would be to limit how many du processes kubelet can spawn at a time.

@msau42 as there exists so many MetricsProvider implements, inject a du counter to [fsResourceAnalyzer](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/server/stats/fs_resource_analyzer.go#L65) will break too many codes.
Do you think it's reasonable to add a global counter in pkg/volume/metrics_du.go?

Reply:
fsResourceAnalyzer check filesystem usage every `volume-stats-agg-period` period(default 1min). IMO du command timeout will terminate last period's du command, although every period's du process number are different.

Reply:
Can we enhance the metrics collector to detect if there's still an operation pending from the last period?  And if so, don't start a new one?

Reply:
What I'm worried about in the timeout case, is for truly large volumes that do take a long time, we would never be able to successfully gather any metrics about it, because it would always timeout.

Reply:
> Can we enhance the metrics collector to detect if there's still an operation pending from the last period? And if so, don't start a new one?

@msau42  done, PTAL